### PR TITLE
Get DBTest passing Assert Status Checked

### PR DESCRIPTION
**Pull request by sike-evolve**
Closes https://github.com/facebook/rocksdb/pull/7737

### CODE REVIEW

File: db/db_test.cc [hunk 1]
```diff
@@ -2546,18 +2550,18 @@ TEST_F(DBTest, PurgeInfoLogs) {
     // For mode (1), test DestroyDB() to delete all the logs under DB dir.
     // For mode (2), no info log file should have been put under DB dir.
     std::vector<std::string> db_files;
-    env_->GetChildren(dbname_, &db_files);
+    ASSERT_TRUE(env_->GetChildren(dbname_, &db_files).IsNotFound());
     for (std::string file : db_files) {
       ASSERT_TRUE(file.find("LOG") == std::string::npos);
     }
```
Comment (mrambacher) [inline db/db_test.cc:2556]: This does not make sense to me.  If the GetChildren is returning NotFound, then the loop over db_files is never executed (nothing to loop on).

File: db/db_test.cc [hunk 2]
```diff
@@ -4119,9 +4137,9 @@ TEST_F(DBTest, WriteSingleThreadEntry) {
   std::vector<port::Thread> threads;
   dbfull()->TEST_LockMutex();
   auto w = dbfull()->TEST_BeginWrite();
-  threads.emplace_back([&] { Put("a", "b"); });
+  threads.emplace_back([&] { ASSERT_OK(Put("a", "b")); });
```
Comment (mrambacher) [inline db/db_test.cc:4140]: I am not sure if this is correct or if it needs to return the Status and someone else has to check it.  I am not sure the semantics of a thread fails/asserts.

**Review by mrambacher — APPROVED** [review]
I am not wild about the set status and return in the db_impl_experimental class, but cannot see an easy way around it.

Otherwise, a few minor comments and looks good.

File: db/db_test.cc [hunk 3]
```diff
@@ -2257,7 +2261,7 @@ TEST_F(DBTest, SnapshotFiles) {
     ReadOptions roptions;
     std::string val;
     for (unsigned int i = 0; i < 80; i++) {
-      stat = snapdb->Get(roptions, cf_handles[i < 40], Key(i), &val);
+      ASSERT_OK(snapdb->Get(roptions, cf_handles[i < 40], Key(i), &val));
```
[the remark this answers is not in the document]
Comment (sike-evolve) [reply db/db_test.cc:2381]: "stat" has been deleted. This "ASSERT_OK(snapdb->Get(roptions, cf_handles[i < 40], Key(i), &val)); " is new added code.

File: db/db_test.cc (hunk 2 above)
Comment (sike-evolve) [reply db/db_test.cc:4140]: "ASSERT_OK" just makes sure "Put" returns the correct status. If not, the test case should show the failure.

File: db/db_test.cc (hunk 1 above)
Comment (sike-evolve) [reply db/db_test.cc:2556]: I agree, the loop is not necessary.

Comment (adamretter) [reply db/db_test.cc]: @sike-evolve Can you remove the redundant loop please?

**Comment by adamretter** [comment]
@mrambacher I think we have addresses your review comments, any chance of a merge?

**Review by adamretter — APPROVED** [review]
LGTM - thanks Si

**Review by pdillinger — CHANGES_REQUESTED** [review]
To confirm this works and continues to work, please remove db_test from TESTS_FAILING_ASC in Makefile. Thanks

**Review by pdillinger — APPROVED** [review]

**Pull request closed without merging**